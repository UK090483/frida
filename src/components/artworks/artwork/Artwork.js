import React from "react"
import Frida from "../../frida/frida"
import styled from "styled-components"
import PropTypes from "prop-types"

import Container from "./container/artworkContainer"
import ArtworkImage from "./artworkImage/artworkImage"
import ArtworkInfo from "./artworkInfo/ArtworkInfo"
import Banner from "./banner/banner"

function Artwork({ artwork, color = "pink", preventClick }) {
  const {
    availability,
    artworkName,
    artistName,
    price,
    image,
    banner,
  } = artwork

  return (
    <Container artwork={artwork} preventClick={preventClick}>
      <ArtworkImage
        alt={`artwork ${artworkName} from ${artistName}`}
        image={image}
        fluid={image.fluid500}
      />
      {banner === "hinzundkunzt" && <Banner></Banner>}
      <ArtistName>
        <Frida text={artistName} textColor={color}></Frida>
      </ArtistName>

      <ArtworkInfo
        availability={availability}
        price={price}
        artworkName={artworkName}
      />
    </Container>
  )
}

const ArtistName = styled.h3`
  margin-top: 10px;
  font-size: 1rem;
  margin-bottom: 5px;
  @media ${({ theme }) => theme.device.tablet} {
    font-size: 1.35rem;
  }
`
ArtistName.defaultProps = {
  theme: {
    device: { tablet: 100 },
  },
}

Artwork.propTypes = {
  artwork: PropTypes.object,
  handleClick: PropTypes.func,
  handleLoaded: PropTypes.func,
}

Artwork.defaultProps = {
  artwork: { price: 99999999999999 },
  handleClick: () => {},
  handleLoaded: () => {},
}

export default Artwork
