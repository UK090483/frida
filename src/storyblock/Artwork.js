// import React from "react"
// import SingleArtwork from "../components/artworks/singleArtwork/singleArtwork"
// import getArtwork from "../components/artworks/helper/getArtwork"
// export default function storyblock(props) {
//   return <SingleArtwork artwork={getArtwork({ content: props.blok })} />
// }
