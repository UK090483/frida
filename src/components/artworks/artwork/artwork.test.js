import React from "react"
import renderer from "react-test-renderer"
import Artwork from "./Artwork"
import { cleanup, fireEvent, render } from "@testing-library/react"
import "jest-styled-components"

describe("Artwork ", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Artwork></Artwork>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
