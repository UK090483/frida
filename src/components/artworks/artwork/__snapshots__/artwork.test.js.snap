// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Artwork  renders correctly 1`] = `
.c2 {
  white-space: nowrap;
  display: inline;
}

.c3 {
  position: relative;
  color: pink;
}

.c3:before {
  content: "";
  border-bottom: rgb(0,0,0) solid 0.16em;
  left: 0;
  right: 0;
  bottom: 0;
  position: absolute;
}

.c0 {
  width: 100%;
  background-clip: padding-box;
  margin-bottom: 80px;
}

.c6 {
  padding-top: 15px;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-height: 10px;
  margin-left: 29px;
}

.c7 {
  position: relative;
}

.c7::after {
  content: "";
  position: absolute;
  top: 50%;
  -webkit-transform: translateY(-50%) translateX(-30px);
  -ms-transform: translateY(-50%) translateX(-30px);
  transform: translateY(-50%) translateX(-30px);
  border-radius: 50%;
  left: 0;
  width: 20px;
  height: 20px;
  background-color: green;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1.35rem;
  font-weight: 500;
}

.c5 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c8 {
  padding-left: 10px;
  padding-top: 15px;
  margin-left: auto;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  text-align: right;
}

.c1 {
  margin-top: 10px;
  font-size: 1rem;
  margin-bottom: 5px;
}

@media 100 {
  .c1 {
    font-size: 1.35rem;
  }
}

<div
  className="c0"
  data-testid="artwork-prev"
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  role="button"
>
  
  <h3
    className="c1"
  >
    <span
      className="c2"
    >
      #Meet
      <span
        className="c3"
        color="#f5c5d9"
      >
        Frida
      </span>
    </span>
  </h3>
  <div
    className="c4"
  >
    <div
      className="c5"
    >
      <div
        className="c6"
      >
        <span
          className="c7"
        >
          prop 
        </span>
        artworkName is missing
      </div>
      <div
        className="c8"
      >
        119047619047620
        â‚¬
      </div>
    </div>
  </div>
</div>
`;
